public/assets/js/publishFunctions.js

```diff
@@ -65,41 +65,87 @@ function validateClaimName (name) {
 	});
 	return deferred;
 }
+// validation function which checks all aspects of the publish submission
+function validateSubmission(stagedFiles, name){
+	var deferred = new Promise(function (resolve, reject) {
+		// make sure only 1 file was selected
+		if (!stagedFiles) {
+			reject(new FileError("Please select a file"));
+		} else if (stagedFiles.length > 1) {
+			reject(new FileError("Only one file is allowed at a time"));
+		}
+		// validate the file's name, type, and size
+		try {
+			validateFile(stagedFiles[0]);
+		} catch (error) {
+			reject(error);
+		}
+		// make sure the claim name has not already been used
+		validateClaimName(name)
+			.then(function() {
+				resolve();
+			})
+			.catch(function(error) {
+				reject(error);
+			})
+	});
+	return deferred;
+}
 
-/* regular publish helper functions */
+/* publish helper functions */
 
+// When a file is selected for publish, validate that file and 
+// stage it so it will be ready when the publish button is clicked.
 function previewAndStageFile(selectedFile){ 
 	var previewHolder = document.getElementById('asset-preview-holder');
 	var dropzone = document.getElementById('drop-zone');
 	var previewReader = new FileReader();
 	var nameInput = document.getElementById('publish-name'); 
-	// validate the file
+	// validate the file's name, type, and size
 	try {
 		validateFile(selectedFile);
 	} catch (error) {
 		showError('input-error-file-selection', error.message);
 		return;
 	}
-	// set the preview
-	if (selectedFile.type === 'video/mp4') {
-		
-	} else {
+	// set the image preview, if a preview was provided
+	if (selectedFile.type !== 'video/mp4') {
 		previewReader.readAsDataURL(selectedFile);
 		previewReader.onloadend = function () {
 			dropzone.style.display = 'none';
 			previewHolder.style.display = 'block';
-			previewHolder.innerHTML = '<img width="100%" src="' + previewReader.result + '" alt="image preview"/>';
-			
+			previewHolder.innerHTML = '<img width="100%" src="' + previewReader.result + '" alt="image preview"/>';			
 		};
 	}
 	// set the name input value to the image name if none is set yet
 	if (nameInput.value === "") {
```

Comment:
@98farhan94 this is where the claim-name input box is filled based off of the file name.  I added a check here to replace spaces with dashes as you suggested (line 123)